docs/aqua-book/language/modularity.md

```diff
@@ -0,0 +1,157 @@
+# Modularity
+
+## Source File Structure
+
+An Aqua source file has a header and a body. The body contains [function definitions](./expressions/functions.md), [services](./services.md), [types](./types.md), [constants](./values.md#constants). The header is dedicated to code management: it specifies [the name of the module](#giving-a-name-to-an-aqua-module-with-aqua), [what is declared by the module](#specifying-what-is-declared-by-the-module-with-declares), [what is exported from the module](#exporting-to-the-host-language-with-export) and [what is imported into the module](#importing-other-modules).
+
+## Giving a name to an Aqua module with `aqua`
+
+**Every** Aqua source file should begin with `aqua` keyword followed by the the name of the aqua module presented by the file.
+
+```aqua
+-- `aqua` expression may only be on the very first line of the file
+aqua AquaFile
+```
+
+Module name can contain dots, e.g. `aqua Aqua.File`.
+
+This name is used when the module is imported with `use`, see [Importing other modules with `use`](#with-use).
+
+## Specifying what is declared by the module with `declares`
+
+The `aqua AquaFile` expression may optionally include a `declares` section. This section enumerates the elements that the module will make available for **other modules that import it**. If the `declares` section is omitted, the module does not declare anything for other modules to use.
+
+```aqua
+-- This module declares `CONST_NAME`, `ServiceName`, `MyType` and `fn`
+aqua AquaFile declares CONST_NAME, ServiceName, MyType, fn
+
+const CONST_NAME = "something"
+
+service ServiceName:
+  do_something()
+  
+data MyType:
+  result: i32  
+
+func fn() -> string:
+  <- CONST_NAME
+```
+
+To declare everything contained in the file, use `declares *`:
+
+```aqua
+-- This module declares `CONST_NAME`, `ServiceName`, `MyType` and `fn`
+aqua AquaFile declares *
+
+const CONST_NAME = "something"
+
+service ServiceName:
+  do_something()
+  
+data MyType:
+  result: i32  
+
+func fn() -> string:
+  <- CONST_NAME
+```
+
+Note that symbols declared with `declares` are not exported to the host language. To export symbols to the host language, use [`export`](#exporting-to-the-host-language-with-export).
+
+## Importing other modules
+
+Aqua modules can import other modules to use their [declarations](#specifying-what-is-declared-by-the-module-with-declares). There are two ways to import a module: with [`import`](#with-import) and [`use`](#with-use).
+
+### With `import`
+
+The main way to import a module is via `import`. Everything declared in the imported module comes into the current namespace directly.
```

Comment:
I would push `use` to the top, as we aim to have `use` the main way to use modules